client.go

```diff
@@ -146,40 +161,41 @@ func (c *Client) Post(out interface{}, entityType string, values url.Values, bod
 	if err != nil {
 		return errors.Wrapf(err, "Error parsing entity type: %s", entityType)
 	}
-	u := c.BaseURL.ResolveReference(rel)
+	u := c.baseURL.ResolveReference(rel)
 
 	if values == nil {
 		values = url.Values{}
 	}
 	values = c.defaultParams(values)
 
-	c.debugLog("[targetprocess] POST %s/%s?%s", c.BaseURL, entityType, values.Encode())
+	c.debugLog("[targetprocess] POST %s/%s?%s", c.baseURL, entityType, values.Encode())
 	fullURL := fmt.Sprintf("%s?%s", u.String(), values.Encode())
 
 	req, err := http.NewRequest("POST", fullURL, bytes.NewBuffer(body))
 	if err != nil {
-		return errors.Wrapf(err, "Invalid POST request: %s/%s", c.BaseURL, entityType)
+		return errors.Wrapf(err, "Invalid POST request: %s/%s", c.baseURL, entityType)
 	}
 	return c.do(out, req, entityType)
 }
 
 func (c *Client) do(out interface{}, req *http.Request, urlPath string) error {
+	noParameterURL := fmt.Sprintf("%s://%s%s", req.URL.Scheme, req.URL.Host, req.URL.Path)
+
 	// Set the headers that will be required for every request
 	req.Header.Add("Content-Type", "application/json")
 	req.Header.Add("Accept", "application/json")
 	if c.UserAgent != "" {
 		req.Header.Add("User-Agent", c.UserAgent)
 	}
-
 	resp, err := c.Client.Do(req)
 	if err != nil {
-		return errors.Wrapf(err, "HTTP request failure on %s/%s", c.BaseURL, urlPath)
+		return errors.Wrapf(err, "HTTP request failure on %s", noParameterURL)
 	}
 
 	// Empty the body and close it to reuse the Transport
 	defer func() {
-		io.Copy(ioutil.Discard, resp.Body) // nolint:golint,errcheck
-		resp.Body.Close()
+		_, _ = io.Copy(ioutil.Discard, resp.Body) // nolint:golint,errcheck
```

Comment:
Is the linter comment necessary? I thought `_, _ :=` would take care of the linter.

Reply:
Ah you know what I didn't think about that when I changed it. My IDE still complained so I did what it told me, but now I think that comment should have stopped the IDE from complaining. I'll revisit.